src/meta/processors/admin/CreateSnapshotProcessor.cpp

```diff
@@ -0,0 +1,100 @@
+/* Copyright (c) 2019 vesoft inc. All rights reserved.
+ *
+ * This source code is licensed under Apache 2.0 License,
+ * attached with Common Clause Condition 1.0, found in the LICENSES directory.
+ */
+
+
+#include "meta/processors/admin/CreateSnapshotProcessor.h"
+#include "meta/processors/admin/SnapShot.h"
+#include "meta/ActiveHostsMan.h"
+
+namespace nebula {
+namespace meta {
+using SignType = storage::cpp2::EngineSignType;
+
+void CreateSnapshotProcessor::process(const cpp2::CreateSnapshotReq& req) {
+    UNUSED(req);
+    auto snapshot = genSnapshotName();
+    folly::SharedMutex::WriteHolder wHolder(LockUtils::snapshotLock());
+
+    auto hosts = ActiveHostsMan::getActiveHosts(kvstore_);
+    if (hosts.empty()) {
+        LOG(ERROR) << "There is no active hosts";
+        resp_.set_code(cpp2::ErrorCode::E_NO_HOSTS);
+        onFinished();
+        return;
+    }
+
+    // step 1 : Let recode the snapshot to meta , default snapshot status is CREATING.
+    //          The purpose of this is to handle the failure of the checkpoint.
+    std::vector<kvstore::KV> data;
+    data.emplace_back(MetaServiceUtils::snapshotKey(snapshot),
+                      MetaServiceUtils::snapshotVal(cpp2::SnapshotStatus::INVALID,
+                                                    NetworkUtils::toHosts(hosts)));
+
+    if (!doSyncPut(std::move(data))) {
+        LOG(ERROR) << "Write snapshot meta error";
+        resp_.set_code(cpp2::ErrorCode::E_STORE_FAILURE);
+        onFinished();
+        clusterCheck_->addUnblockingTask();
+        return;
+    }
+
+    // step 2 : Blocking all writes action for storage engines.
+    auto signRet = Snapshot::instance(kvstore_)->blockingWrites(SignType::BLOCK_ON);
```

Comment:
What if some parts have blocked before, and we fail here, then user retry, but this time we block all partitions. Because you unblock in async way by `addUnblockingTask`, is it possible that after we block at second time, and then get unblocked by first one.

Reply:
make sense ! Let me see how to solve the riddle. Any idea ？

Reply:
> What if some parts have blocked before, and we fail here, then user retry, but this time we block all partitions. Because you unblock in async way by `addUnblockingTask`, is it possible that after we block at second time, and then get unblocked by first one.

Which brings me to another question,I also need to check unblocking task is running when create checkpoint , because the unblocking task is cycle execute until success. If this task is being performed, create checkpoint will be rejected.